x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -343,6 +343,7 @@ export const createLifecycleExecutor =
             : { index: { _id: event[ALERT_UUID]! } },
           event,
         ]),
+        refresh: 'wait_for',
```

Comment:
We should do the same for persistent alerts => x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts.

I can see the refresh is configurable (parameter), let's make sure it guarantees to `wait_for` whenever it's a rule running. We may have to remove the parameter and make it `wait_for` permanently..

Reply:
Yes, i saw security team made it configurable and didn't want to modify their code but let me go over it one more time.

Reply:
Sounds good, we may need to track this for future reference. We'll most likely develop other features relying on alerts-as-data being persisted and could encounter odd race conditions. Luckily the current implementation passes `wait_for` as a value when `actions.length > 0` but other features may not be action related.

Reply:
Yes, it looks like the only usage of it is in bulkCreate and it sets refresh 'wait_for' when `actions.length` is truthy.